### PR TITLE
Switch to DefinitelyTyped types

### PR DESCRIPTION
**Pull request by bovine3dom**
These seem to be updated more regularly and generally more comprehensive than web-ext-types.

### CODE REVIEW

File: src/background.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ browser.runtime.onStartup.addListener(_ => {
         } else {
             native.run("hostname").then(hostname => {
                 for (const host of hosts) {
-                    if ((new RegExp(host)).exec(hostname.content)) {
+                    if (new RegExp(host).exec(hostname.content)) {
```
Comment (bovine3dom) [inline src/background.ts:99]: Does `new` really bind this tightly? I think this is a bad prettification rule as it is very un-obvious to me.

File: src/lib/messaging.ts [hunk 2]
```diff
@@ -79,23 +83,80 @@ function backgroundHandler<
...
+    // ./src/lib/messaging.ts
+    // [tsl] ERROR in /home/olie/projects/tridactyl/src/lib/messaging.ts(115,40)
+    //       TS2558: Expected 0 type arguments, but got 2.
+    // ```
+    //
+    // don't you understand?
+    //
+    // (This is why there is an `: any => `)
+    browser.runtime.onMessage.addListener(
+        (message: any, sender: browser.runtime.MessageSender): any => {
```
Comment (bovine3dom) [inline src/lib/messaging.ts:146]: @farnoy I think this function is yours. Do you know why DefinitelyTyped hates it so much?

**Comment by bovine3dom** [comment]
@glacambre I don't understand why the unit tests are failing:

```
 FAIL  src/lib/mod.test.ts
  ● Test suite failed to run

    TypeError: src/lib/mod.test.ts: Emit skipped for language service

      at compileFn (node_modules/ts-jest/dist/compiler/language-service.js:144:23)
      at Object.compile (node_modules/ts-jest/dist/compiler/instance.js:52:21)
      at TsJestTransformer.process (node_modules/ts-jest/dist/ts-jest-transformer.js:85:41)
      at ScriptTransformer.transformSource (node_modules/@jest/transform/build/ScriptTransformer.js:481:35)
      at ScriptTransformer._transformAndBuildScript (node_modules/@jest/transform/build/ScriptTransformer.js:586:40)
      at ScriptTransformer.transform (node_modules/@jest/transform/build/ScriptTransformer.js:624:25)
```

any ideas?

File: tsconfig.json [hunk 3]
```diff
@@ -7,7 +7,6 @@
     "sourceMap": true,
     "target": "es2017",
     "lib": ["es2017","es2019.array", "es2018.promise", "dom", "dom.iterable"],
-    "types": ["web-ext-types"],
```
Comment (glacambre) [inline tsconfig.json:10]: Maybe this should be an empty array? The error makes it sound like jest failed to translate the test files from ts to js, which to me means that there's probably an issue with the ts config.

Comment (bovine3dom) [reply tsconfig.json:10]: No luck - that breaks the build (Typescript automatically uses all `@types` folders unless you specify anything there) and doesn't fix the test either.

Logically, manually specifying `@types/firefox-webext-browser` doesn't work either.

File: src/lib/messaging.ts (hunk 2 above)
Comment (farnoy) [reply src/lib/messaging.ts:146]: @bovine3dom Sorry about the delay. The callback to `addListener` should return a boolean or a Promise, but MDN docs advise against using `async` closure here directly. I think we need to modify `backgroundHandler`, or do something like `return !!backgroundHandler(root, message, sender)` to cast this to a boolean

Comment (farnoy) [reply src/lib/messaging.ts:146]: Actually `return Promise.resolve(backgroundHandler(root, message, sender))` should be enough, but I'm not sure if it's going to typecheck

Comment (bovine3dom) [reply src/lib/messaging.ts:146]: That seems to work, thanks - 27c5a036b424020d49b9a1bf13206ca753471b4d

**Pull request merged**